odbc-api/src/connection.rs

```diff
@@ -159,7 +159,12 @@ impl<'c> Connection<'c> {
         sleep: impl Sleep,
     ) -> Result<Option<CursorPolling<StatementImpl<'_>>>, Error> {
         let query = SqlText::new(query);
-        let lazy_statement = move || self.allocate_statement();
+        let lazy_statement = move || {
+            self.allocate_statement().and_then(|mut stmt| {
+                stmt.set_async_enable(true).into_result(&stmt)?;
```

Comment:
flyby: shouldn't this implementation set the statement to be async?

Reply:
good catch: Fixed in 8.1.2